docker/server/src/main/common/Dockerfile

```diff
@@ -0,0 +1,44 @@
+ARG BASE
+FROM $BASE as install
+
+COPY licenses/ /
+
+# Note: all of the pip installs have the --no-index flag, with the expectation that all external dependencies are
+# already satisfied via the base image. If that is not the case, we want the install to error out, and we'll need to
+# update the base image with the extra requirements before proceeding here.
+
+COPY requirements.txt requirements.txt
+RUN set -eux; \
+    python3 -m pip install -q --no-index --no-cache-dir -r requirements.txt; \
+    rm requirements.txt
+
+COPY deephaven-jpy-wheel/ /deephaven-jpy-wheel
+RUN set -eux; \
+    python3 -m pip install -q --no-index --no-cache-dir /deephaven-jpy-wheel/*.whl; \
+    rm -r /deephaven-jpy-wheel
+
+COPY deephaven-wheel/ /deephaven-wheel
+RUN set -eux; \
+    python3 -m pip install -q --no-index --no-cache-dir /deephaven-wheel/*.whl; \
+    rm -r /deephaven-wheel
+
+COPY deephaven2-wheel/ /deephaven2-wheel
+RUN set -eux; \
+    python3 -m pip install -q --no-index --no-cache-dir /deephaven2-wheel/*.whl; \
+    rm -r /deephaven2-wheel
+
+ARG DEEPHAVEN_VERSION
+
+ADD serverApplicationDist/server-${DEEPHAVEN_VERSION}.tar /opt/deephaven
+
+RUN set -eux; \
+    ln -s /opt/deephaven/server-${DEEPHAVEN_VERSION} /opt/deephaven/server
+
+COPY image-bootstrap.properties image-bootstrap.properties
+
+FROM $BASE
+COPY --from=install / /
+VOLUME /data
+VOLUME /cache
+HEALTHCHECK --interval=3s --retries=3 --timeout=11s CMD /bin/grpc_health_probe -addr=localhost:8080 -connect-timeout=10s || exit 1
```

Comment:
irritating that we can't just FROM this in, and will have to change the settings in more than one place if we alter our expectations

Reply:
I think we can generalize a bit in the future, and create a wrapper script around `grpc_health_probe` that does the right thing against the configurations we set (TLS, timeouts, etc).